modules/raft/src/main/java/org/apache/ignite/internal/raft/RaftGroupServiceImpl.java

```diff
@@ -679,6 +694,14 @@ private Peer randomNode(@Nullable Peer excludedPeer) {
 
         int lastPeerIndex = excludedPeer == null ? -1 : peers0.indexOf(excludedPeer);
 
+        if (peers0.size() == 1) {
+            if (lastPeerIndex != -1) {
```

Comment:
Why do we even need to check this if the peer list has a single element?

Reply:
Well it is kind of a Fast Path, as we already have `indexOf`, so we don't need to call method, right? But yes, it looks too bizarre, I removed it